packages/connection-form/src/hooks/use-connect-form.ts

```diff
@@ -206,9 +206,14 @@ function buildStateFromConnectionInfo(
   // Only enable connection string editing (and in particular, doing so in
   // protected connection strings mode) when it's the default connection
   // string and the connection has not been connected to (saved recent/favorite).
-  const isNewDefaultConnection =
+  const isDefaultCnnStr =
+    initialConnectionInfo.connectionOptions.connectionString ===
+      defaultConnectionString ||
     initialConnectionInfo.connectionOptions.connectionString ===
-      defaultConnectionString && !initialConnectionInfo.lastUsed;
+      `${defaultConnectionString}/`;
```

Comment:
Might as well do this, I guess?

```js
  let isDefaultCnnStr = false;
  try {
    isDefaultCnnStr = 
      new ConnectionString(initialConnectionInfo.connectionOptions.connectionString).toString() ===
      new ConnectionString(defaultConnectionString).toString();
  } catch { /* ignore */ }
```

Reply:
I'm not a big fan of using toString to compare instances of objects because I come from a Java world, as toString should be used only for displaying an object, not comparing.

I'll do the change because I suppose it is fine to do this here.

Reply:
Ah, I've seen that it doesn't have the same behaviour as the previous code, because the ConnectionStringInput should support invalid connection strings in the comparison.

Reply:
> I'm not a big fan of using toString to compare instances of objects because I come from a Java world, as toString should be used only for displaying an object, not comparing.

I mean, this isn't really about `.toString()`, you could as well use `.href`. The point is more that `x => new ConnectionString(x).toString()` is a good normalization function for connection strings. This PR adds handling for trailing slashes, but you could basically go arbitrarily deep on this, add support for percent encoding normalization, etc. until you've built a full URL parser, so why not just use an actual URL parser?

> it doesn't have the same behaviour as the previous code

In which cases would the behavior differ?

Reply:
The tests for ConnectionStringInput fail, I'm still investigating the root cause.

I'm fine using the ConnectionString with href if it normalises, I'm not sure it will have the exact same behaviour, but I'm investigating what it does. At the end, we don't want to have false positives, like for example:

```
"mongodb://user:pwd@test.com/ === mongodb://test.com"
```

That is why I was using strings on the first place.

Reply:
> At the end, we don't want to have false positives, like for example:

Yeah, no, nothing to worry about here. :) This is just like doing `new URL(a).href === new URL(b).href` for regular URLs.